packages/dashboard/src/components/widgets/widget.tsx

```diff
@@ -33,50 +29,29 @@ const WidgetComponent: React.FC<WidgetProps> = ({
 }) => {
   const { x, y, z, width, height } = widget;
 
-  // TODO: Replace with Redux dispatch
-  const [assets, setAssets] = useState<null | AssetQuery>(null);
-  const [internalWidget, setInternalWidget] = useState(widget);
-
-  const [, drop] = useDrop(
-    () => ({
-      accept: ItemTypes.ResourceExplorerAssetProperty,
-      drop: ({ queryAssetsParam }: { queryAssetsParam: AssetQuery }) => {
-        setAssets(queryAssetsParam);
-      },
-    }),
-    []
-  );
-
-  useEffect(() => {
-    const nextInternalWidget = structuredClone(widget);
-    if (assets) {
-      nextInternalWidget.assets = assets as any;
-    }
-    setInternalWidget(nextInternalWidget);
-  }, [JSON.stringify(widget), JSON.stringify(assets)]);
-
   return (
-    <div
-      ref={drop}
-      {...gestureable('widget')}
-      className={`widget ${readOnly ? 'widget-readonly' : ''}`}
-      style={{
-        zIndex: z.toString(),
-        top: `${cellSize * y}px`,
-        left: `${cellSize * x}px`,
-        width: `${cellSize * width}px`,
-        height: `${cellSize * height}px`,
-      }}
-    >
-      <DynamicWidgetComponent
-        readOnly={readOnly}
-        query={query}
-        viewport={viewport}
-        widget={internalWidget}
-        isSelected={isSelected}
-        widgetsMessages={messageOverrides.widgets}
-      />
-    </div>
+    <WidgetDropTarget widget={widget}>
```

Comment:
I think this should be one of the options that dynamic widget renders. Text widgets are one of the options Dynamic widget renders but shouldn't be a drop target.